mutexbot-reserve/action.yml

```diff
@@ -0,0 +1,48 @@
+---
+name: "MutexBot Reserve"
+description: "Reserve a resource on MutexBot"
+inputs:
+  api_key:
+    description: "API Key"
+    required: true
+  resource_name:
+    description: "Name of the resource to reserve"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: "Reserve resource"
+      shell: bash
+      env:
+        API_KEY: ${{ inputs.api_key }}
+        RESOURCE_NAME: ${{ inputs.resource_name }}
+        GITHUB_REPO: ${{ github.repository }}
+        RUN_ID: ${{ github.run_id }}
+      run: |
+        while true; do
+          RESPONSE=$(curl -s -o response.json -w "%{http_code}" -X POST "https://mutexbot.com/api/resources/global/${RESOURCE_NAME}/reserve" \
+            -H "X-API-KEY: ${API_KEY}" \
+            -H "Content-Type: application/json" \
+            -d "{
+              \"notes\": \"Reserved by https://github.com/${GITHUB_REPO}/actios/runs/${RUN_ID}\"
+            }")
+          if [ "$RESPONSE" -eq 200 ]; then
+            echo "Resource reserved successfully."
+            break
+          elif [ "$RESPONSE" -eq 409 ]; then
+            echo "Resource already reserved, retrying..."
+            sleep 5  # Wait before retrying
```

Comment:
I wonder if the response includes the time when the resource will be available. If so, we can sleep for a certain amount of time

Reply:
It does not sadly